server/app.js

```diff
@@ -230,6 +230,34 @@ function getRelatedObjects (objectID) {
     .catch((error) => console.error(error.message))
 }
 
+const getDistance = (from, to) => {
+  const fromKeys = Object.keys(from)
+  const toKeys = Object.keys(to)
+  const commonKeys = fromKeys.filter(key => toKeys.includes(key))
+
+  const descriptorKeys = MORE_LIKE_THIS_FIELDS.map((field) => {
+    if (field.match(/(.*)[_-]\*$/)) {
+      return field.slice(0, field.length - 1)
+    } else if (field.match(/(.*)[\.]\*$/)) {
+      return field.slice(0, field.length - 2)
+    }
+    return field
+  })
+
+  const distanceKeys = descriptorKeys.filter(descriptorKey => commonKeys.some(commonKey => commonKey.indexOf(descriptorKey) === 0))
+
+  const distance = distanceKeys.reduce((sum, key) => {
+    let absoluteDistance = 0
+    switch(key) {
+      case 'people': absoluteDistance = from[key] === to[key] ? 0 : 100; break;
+      default: absoluteDistance = parseFloat(to[key]) - parseFloat(from[key]); break;
+    }
+    return sum + Math.pow(absoluteDistance, 2)
+  }, 0)
+
+  return distance / distanceKeys.length
```

Comment:
do we need to catch a possible divide by 0 error here?

Reply:
In that case, should I return `Infinity`?